lopper/assists/domain_access.py

```diff
@@ -40,6 +40,65 @@ def check_bit_set(n, k):
 
     return False
 
+# tree: is the lopper system device-tree
+def domain_get_subnodes(tree):
+    try:
+        domain_node = tree['/domains']
+    except:
+        domain_node = None
+
+    direct_node_refs = []
+
+    if domain_node:
+        for node in domain_node.subnodes():
+            # 1) memory access/node = <> nodes
+            try:
+                mem_node = node["memory"].value
+                direct_node_refs.append( node )
+            except:
+                pass
+            # 2) direct access = <> nodes
+            a_nodes = lopper_lib.node_accesses( tree, node )
+            if a_nodes:
+                direct_node_refs.append( node )
+
+    # Remove duplicate entries
+    direct_node_refs = list(dict.fromkeys(direct_node_refs))
+    return direct_node_refs
+
+# tree: is the lopper system device-tree
+def get_mem_nodes(tree):
+    try:
+        memory_nodes = [node for node in tree.nodes('/memory@.*')]
+    except:
+        memory_nodes = []
+
+    return memory_nodes
+
+# node: is the domain node number
+# mem_val: Memory node address and size value to be updated
+# This api takes the memory value(address and size) and creates
+# a new memory node value for memory node reg property based on the
+# address-cells and size-cells property.
+def update_mem_node(node, mem_val):
+    ac = node.parent['#address-cells'].value[0]
+    sc = node.parent['#size-cells'].value[0]
```

Comment:
If you want to protect yourself from inputs that come via yaml/other formats (not that it is common), and also abstract the property a bit, these calls can also just be:

ac = node.parent['#address-cells'][0]

And let the smart indexing get you the first value. (since you aren't looking to have value as a list in your use case anyway).